artemis-commons/src/test/java/org/apache/activemq/artemis/utils/ByteUtilTest.java

```diff
@@ -39,6 +39,15 @@
 
    private static Logger log = Logger.getLogger(ByteUtilTest.class);
 
+   @Test
+   public void testBytesToLong() {
+      long randomA = RandomUtil.randomLong();
+      byte[] randomABytes = ByteUtil.longToBytes(randomA);
+      long randomAOutput = ByteUtil.bytesToLong(randomABytes);
+
+      Assert.assertEquals(randomA, randomAOutput);
```

Comment:
It's not about endinannes.. it was about the signal.

I don't remember what was the case that broke the >>> versus >>

The right thing is definitely to use >>>


This is just about comparing expected results... I'm adding another tests based on bytesToInt from the same test.

Reply:
I realise it wasnt about endianness, that was merely giving an example relating to the comment that doing a roundtrip comparison though 2 methods generally not being a great test, a potentially compensating change that could happen in both sides and which the roundtrip test wouldnt detect issue with despite it likely totally breaking intended uses of the methods. Something a 'specific known input, specific known output' test would detect and fail immediately upon, unless updated because the change was expected and methods tests were similarly being updated accordingly.

Using >>> or >> are both 'right' if they are being used appropriately. I still dont think the previous change to the class made any difference given how the shifts and intermediate results were being used (not storing full shifted values, never referencing shifted-in bits that could be different).